pkg/agent/baker_test.go

```diff
@@ -153,25 +190,71 @@ var _ = Describe("Assert generated customData and cseCmd", func() {
 		}
 
 		expectedCustomData, err := ioutil.ReadFile(fmt.Sprintf("./testdata/%s/CustomData", folder))
-		if err != nil {
-			panic(err)
-		}
+		Expect(err).To(BeNil())
 		Expect(customData).To(Equal(string(expectedCustomData)))
 
 		// CSE
-		cseCommand := baker.GetNodeBootstrappingCmd(config)
+		cseCommand := baker.GetNodeBootstrappingCmd(configCseClone.(*datamodel.NodeBootstrappingConfiguration))
 		if generateTestData() {
 			ioutil.WriteFile(fmt.Sprintf("./testdata/%s/CSECommand", folder), []byte(cseCommand), 0644)
 		}
 		expectedCSECommand, err := ioutil.ReadFile(fmt.Sprintf("./testdata/%s/CSECommand", folder))
-		if err != nil {
-			panic(err)
-		}
+		Expect(err).To(BeNil())
 		Expect(cseCommand).To(Equal(string(expectedCSECommand)))
 
-	}, Entry("AKSUbuntu1604 with k8s version less than 1.18", "AKSUbuntu1604+K8S115", "1.15.7", nil),
-		Entry("AKSUbuntu1604 with k8s version 1.18", "AKSUbuntu1604+K8S118", "1.18.2", nil),
-		Entry("AKSUbuntu1604 with k8s version 1.17", "AKSUbuntu1604+K8S117", "1.17.7", nil),
+		files, err := getDecodedFilesFromCustomdata(customDataBytes)
+		Expect(err).To(BeNil())
+
+		vars, err := getDecodedVarsFromCseCmd([]byte(cseCommand))
+		Expect(err).To(BeNil())
+
+		result := &nodeBootstrappingOutput{
+			customData: customData,
+			cseCmd:     cseCommand,
+			files:      files,
+			vars:       vars,
+		}
+
+		if validator != nil {
+			validator(result)
+		}
+
+	}, Entry("AKSUbuntu1604 with k8s version less than 1.18", "AKSUbuntu1604+K8S115", "1.15.7", func(config *datamodel.NodeBootstrappingConfiguration) {
+		config.KubeletConfig["--dynamic-config-dir"] = "/var/lib/kubelet/"
+	}, func(o *nodeBootstrappingOutput) {
+		etcDefaultKubelet := o.files["/etc/default/kubelet"].value
+
+		Expect(o.vars["KUBELET_FLAGS"]).NotTo(BeEmpty())
+		Expect(strings.Contains(o.vars["KUBELET_FLAGS"], "DynamicKubeletConfig")).To(BeTrue())
+		Expect(strings.Contains(o.vars["KUBELET_FLAGS"], "DynamicKubeletConfig")).To(BeTrue())
+		Expect(strings.Contains(o.vars["KUBELET_FLAGS"], "--dynamic-config-dir")).To(BeFalse())
```

Comment:
Is there some sort of list of all deprecated flags per version that we could check for? I assume similar error could happen with other flags in the future as well.